src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -175,10 +175,12 @@ class TezosPlatformDiscoveryOperations(
         if (!cacheExpired(last)) {
           IO.pure(ent)
         } else {
-          for {
+          (for {
+            _ <- caching.putEntities(networkName, ent)
             updatedEntities <- IO.fromFuture(IO(metadataOperations.runQuery(preCacheEntities)))
             _ <- caching.putAllEntities(updatedEntities)
-          } yield updatedEntities(CacheKey(networkName)).value
+          } yield ()).unsafeRunAsyncAndForget()
+          IO.pure(ent)
```

Comment:
Trying to understand if I got it correctly:
is it essentially returning immediately, while the cache is updated in another thread? 

Shouldn't the code call `IO.contextShift` before running the async IO, to have it run on a different thread?

In any case the `IO.pure(ent)` is returned... hence a single `if` clause (side-effecting) is needed, with code to refresh the case, and then we should return the value anyway, something like

```scala
if (cacheExpired(last) {
  for { ... }.unsafeRunAsyncAndThrowItInTheToilet()
}
IO.pure(ent) // or, equivalently: ent.pure
```